spec/integ/crypto/megolm-backup.spec.ts

```diff
@@ -322,6 +324,22 @@ describe.each(Object.entries(CRYPTO_BACKENDS))("megolm-keys backup (%s)", (backe
             // tell Alice to trust the dummy device that signed the backup
             await waitForDeviceList();
             await aliceCrypto.setDeviceVerified(testData.TEST_USER_ID, testData.TEST_DEVICE_ID);
+
+            importMockImpl = jest
+                .fn()
+                .mockImplementation((keys: IMegolmSessionData[], version: String, opts?: ImportRoomKeysOpts) => {
+                    // need to report progress
+                    if (opts?.progressCallback) {
+                        opts.progressCallback({
+                            stage: "load_keys",
+                            successes: keys.length,
+                            failures: 0,
+                            total: keys.length,
+                        });
+                    }
+                });
+            // @ts-ignore - mock a private method for testing purpose
+            aliceCrypto.importBackedUpRoomKeys = importMockImpl;
```

Comment:
we should definitely not do this. Doing so turns all of these tests from integration tests (where we test the whole stack, including the rust bits) into unit tests (where we are just testing the behaviour of `MatrixClient`).

To be honest, it is a problem that we have existing tests that mock out `CryptoApi.importBackedUpRoomKeys` but which claim to be integration tests, but that's a problem for another day.